core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/compression/GzipCompressor.java

```diff
@@ -38,22 +38,39 @@ public String compressorType() {
     }
 
     @Override
-    public InputStream compress(InputStream inputStream) {
+    public byte[] compress(byte[] content) {
         try {
-            byte[] content = IoUtils.toByteArray(inputStream);
             ByteArrayOutputStream compressedOutputStream = new ByteArrayOutputStream();
             GZIPOutputStream gzipOutputStream = new GZIPOutputStream(compressedOutputStream);
             gzipOutputStream.write(content);
             gzipOutputStream.close();
```

Comment:
Can we use finally and do a    closeQuietly(writer); you can refer closing in  SimpleGeneratorTask

Reply:
`SimpleGeneratorTask`'s `closeQuietly(writer)` is in the `codegen` module, using `closeQuietly(AutoCloseable is, Logger log)` from `IoUtils` in `utils` module